directories/src/components/storage/index.js

```diff
@@ -1,33 +1,115 @@
+  <Input
+    style={{ width: 1000, paddingBottom: 20 }}
+    onChange={onChange}
+    label={{ tag: true, content: "Genesis Hash" }}
+    labelPosition="right"
+    placeholder=""
+  />
+);
+
+const NumberofChunksInput = onChange => (
+  <Input
+    onChange={onChange}
+    style={{ paddingRight: 50, paddingBottom: 50 }}
+    label={{ tag: true, content: "Number of Chunks" }}
+    labelPosition="right"
+    placeholder=""
+  />
+);
+
 class Storage extends Component {
-  componentDidMount() {
-    const { startAppFn } = this.props;
-    // startAppFn();
+  constructor(props) {
+    super(props);
+    this.state = { genesisHash: "", numberOfChunks: 0 };
+  }
+
+  onGenesisHashChange = (e, data) => {
+    this.setState({ genesisHash: data.value });
+  };
+
+  onNumberOfChunksChange = (e, data) => {
+    this.setState({ numberOfChunks: parseInt(data.value) });
+  };
+
+  renderProgress(current, max) {
+    if (current > -1) {
+      return (
+        <div>
+          {current}/{max}
+        </div>
+      );
+    } else {
+      return null;
+    }
+  }
+
+  async onClick() {
+    const { startAppFn, startNode, findTreasure } = this.props;
+
+    const generatedMap = datamap.generate(
+      this.state.genesisHash,
+      this.state.numberOfChunks
+    );
+
+    const transformedMap = _.toArray(generatedMap)
+      .map((value, index) => ({
+        address: value,
+        chunkIdx: index
+      }))
+      .map(obj => findTreasure(obj));
+
+    Promise.all(transformedMap); //TODO better implementation
   }
 
   render() {
-    const { statuses } = this.props;
+    const { statuses, treasures, numberOfCalls } = this.props;
     return (
-      <div className="container">
-        <img src={LOGO} width="100" />
-        <div className="status-container">{statuses.map(s => <p>{s}</p>)}</div>
-      </div>
+      <Container style={{ backgroundColor: "#0267ea" }}>
+        <div style={{ padding: 50 }}>
+          <Header as="h1" style={{ color: "#ffffff" }}>
+            <Image src={LOGO} /> Oyster Toolbox{" "}
+            {this.renderProgress(numberOfCalls, this.state.numberOfChunks)}
+          </Header>{" "}
+          <div style={{ paddingTop: 50 }}>
+            {" "}
+            {GenesisHashInput(this.onGenesisHashChange)}
+            {NumberofChunksInput(this.onNumberOfChunksChange)}
+            <Button onClick={() => this.onClick()}>Look for treasures</Button>
+          </div>
+          {treasures.length != 0 ? TreasureTable(treasures) : null}
+        </div>
+      </Container>
     );
   }
 }
 
 const mapStateToProps = state => ({
-  statuses: state.pow.statuses
+  statuses: state.pow.statuses,
+  treasures: state.test.treasures,
+  numberOfCalls: state.test.numberOfCalls //TODO remove for production
 });
 
 const mapDispatchToProps = dispatch => ({
-  startAppFn: () => dispatch(appActions.startApp())
+  startAppFn: () => dispatch(appActions.startApp()),
+  startNode: () => dispatch(nodeActions.addBrokerNode("asdasd")),
```

Comment:
`addBrokerNode("asdasd")),`   

Just wanted to check and make sure this was intentional?

Reply:
Actually no, I have no idea why that is there. Removed that.